certification/internal/authn/keychain.go

```diff
@@ -0,0 +1,81 @@
+package authn
+
+import (
+	"os"
+	"sync"
+
+	"github.com/docker/cli/cli/config"
+	"github.com/docker/cli/cli/config/types"
+	craneauthn "github.com/google/go-containerregistry/pkg/authn"
+	"github.com/google/go-containerregistry/pkg/name"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+type preflightKeychain struct {
+	mu sync.Mutex
+}
+
+var PreflightKeychain craneauthn.Keychain = &preflightKeychain{}
+
+func (k *preflightKeychain) Resolve(target craneauthn.Resource) (craneauthn.Authenticator, error) {
```

Comment:
This needs a comment detailing the "why" and "how".

Reply:
Done. LMKWYT